README.md

```diff
@@ -23,16 +46,85 @@ or by editing your `go.mod` file as [described by the gomod documentation](https
 
 ## Usage
 
+WebSockets are one of the default transports enabled when constructing a standard libp2p
+Host, along with [TCP](https://github.com/libp2p/go-tcp-transport).
+
+Calling [`libp2p.New`][godoc-libp2p-new] to construct a libp2p Host will enable
+the WebSocket transport, unless you override the default transports by passing in
+`Options` to `libp2p.New`.
+
+To explicitly enable the WebSocket transport while constructing a host, use the
+`libp2p.Transport` option, passing in the `ws.New` constructor function:
+
+``` go
+
+import (
+    "context"
+
+    libp2p "github.com/libp2p/go-libp2p"
+    ws "github.com/libp2p/go-ws-transport"
+)
+
+ctx := context.Background()
+
+// WebSockets only:
+h, err := libp2p.New(ctx,
+    libp2p.Transport(ws.New)
+)
+```
+
+The example above will replace the default transports with a single WebSocket
+transport. To add multiple tranports, use `ChainOptions`:
+
+``` go
+// WebSockets and QUIC:
+h, err := libp2p.New(ctx,
+    libp2p.ChainOptions(
+        libp2p.Transport(ws.New),
+        libp2p.Transport(quic.NewTransport)) // see https://github.com/libp2p/go-libp2p-quic-transport
+)
+```
+
+## Addresses
+
+The WebSocket transport supports [multiaddrs][multiaddr] that contain a `ws`
+component, which is encapsulated within (or layered onto) another valid TCP
+multiaddr.
+
+Examples:
+
+| addr                          | description                                        |
+|-------------------------------|----------------------------------------------------|
+| `/ip4/1.2.3.4/tcp/1234/ws`    | IPv4: 1.2.3.4, TCP port 1234                       |
+| `/ip6/::1/tcp/1234/ws`        | IPv6 loopback, TCP port 1234                       |
+| `/dns4/example.com/tcp/80/ws` | DNS over IPv4, hostname `example.com`, TCP port 80 |
+
+Notice that the `/ws` multiaddr component contextualizes an existing TCP/IP
+multiaddr and does not require any additional addressing information.
+
+## Security and Multiplexing
+
+While the WebSocket spec defines a `wss` URI scheme for encrypted WebSocket
+connections, support for `wss` URIs relies on TLS, which wraps the WebSocket
+connection in a similar manner to TLS-protected HTTP traffic.
+
+As libp2p does not integrate with the TLS Certificate Authority infrastructure
+by design, security for WebSockets is provided by a [transport
+upgrader][transport-upgrader]. The transport upgrader negotiates transport
+security for each connection according to the protocols supported by each party.
+
+The transport upgrader also negotiates a stream multiplexing protocol to allow
+many bidirectional streams to coexist on a single WebSocket connection.
 
```

Comment:
It would also be nice to add a note about proxying. Many HTTP(s) proxies natively support websockets so they're a way to punch out of corporate networks.

Reply:
That's a good call. I'll add that in on the next round.